src/interaction-handlers/sokoban.ts

```diff
@@ -0,0 +1,90 @@
+import { LanguageKeys } from '#lib/i18n/LanguageKeys';
+import {
+	buildGameControls,
+	buildSokobanGameFromResolvableLevel,
+	buildSokobanGameFromVisualLevel,
+	Direction,
+	encodeResolvableLevel
+} from '#lib/utilities/sokoban';
+import { ActionRowBuilder, ButtonBuilder } from '@discordjs/builders';
+import { InteractionHandler } from '@skyra/http-framework';
+import { getSupportedLanguageT } from '@skyra/http-framework-i18n';
+import { ButtonStyle, MessageFlags, type APIButtonComponentWithCustomId } from 'discord-api-types/v10';
+
+const Root = LanguageKeys.Commands.Sokoban;
+
+export class UserHandler extends InteractionHandler {
+	public async run(interaction: InteractionHandler.ButtonInteraction, [directionOrRetry, startTimestampOrEncodedLevel, moves]: Parameters) {
+		const t = getSupportedLanguageT(interaction);
+		// if the restart button is pressed, reset the current level
+		if (directionOrRetry === 'retry') {
+			const encodedLevel = startTimestampOrEncodedLevel!.replaceAll('-', '.');
+			const levelResult = buildSokobanGameFromResolvableLevel(encodedLevel);
+			if (levelResult.isErr()) {
+				return interaction.update({
+					content: t(LanguageKeys.Commands.Sokoban.SokobanInvalidComponent, { value: levelResult.unwrapErr() }),
+					flags: MessageFlags.Ephemeral
+				});
+			}
```

Comment:
Can we ever trigger this error? Considering it's been validated by Artiel earlier, I think we can just `unwrap()` this one
```suggestion
			const level = buildSokobanGameFromResolvableLevel(encodedLevel).unwrap();
```